vsphere_guest reconfigure vm_floppy

dericcrago opened the pull request:
##### SUMMARY
Allow user to reconfigure vm_floppy using vsphere_guest.

Current vm_floppy configuration is only at VM creation, this adds it to reconfigure as well.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
cloud/vmware/vsphere_guest (specifically reconfiguration)

##### ANSIBLE VERSION
```
ansible 2.5.0 (vsphere_guest_reconfigure_vm_floppy f3244d7b1e) last updated 2018/02/01 12:59:11 (GMT -400)
  config file = /Users/deric/.ansible.cfg
  configured module search path = [u'/Users/deric/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/deric/ansible/lib/ansible
  executable location = /Users/deric/ansible/bin/ansible
  python version = 2.7.14 (default, Sep 25 2017, 09:54:19) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
```

##### ADDITIONAL INFORMATION
```yaml
# before
- name: Reconfigure removable devices to 'client'
  vsphere_guest:
    vcenter_hostname: vcenter
    username: vcenter_username
    password: vcenter_password
    guest: inventory_hostname_short
    esxi:
      datacenter: datacenter
      hostname: esxi_host
    vm_hardware:
      vm_floppy:
        type: client
      vm_cdrom:
        type: client
    state: reconfigured
  delegate_to: localhost
# changed: [inventory_hostname_short -> localhost] => {"changed": true, "changes": {"cdrom": {"type": "client"}}, "failed": false}

# after
- name: Reconfigure removable devices to 'client'
  vsphere_guest:
    vcenter_hostname: vcenter
    username: vcenter_username
    password: vcenter_password
    guest: inventory_hostname_short
    esxi:
      datacenter: datacenter
      hostname: esxi_host
    vm_hardware:
      vm_floppy:
        type: client
      vm_cdrom:
        type: client
    state: reconfigured
  delegate_to: localhost
# changed: [inventory_hostname_short -> localhost] => {"changed": true, "changes": {"cdrom": {"type": "client"}, "floppy": {"type": "client"}}, "failed": false}
```

dericcrago committed:
added 'get_floppy_params'

dericcrago committed:
added 'vm_floppy' to reconfigure

Akasurde commented:
As per [meeting decision](https://meetbot.fedoraproject.org/ansible-vmware/2018-02-05/ansible_vmware_working_group_meeting.2018-02-05-15.59.html), we decided to merge to PR provided with playbook `-vvvv` output.

Could you please add `-vvvv` output and we can take it from there ?

dericcrago commented:
Hi @Akasurde, here is the '-vvvv' output as requested. Thanks!

```
$ ansible-playbook -i hosts.yml reconfigure_vm_floppy.yml -vvvv
ansible-playbook 2.5.0 (vsphere_guest_reconfigure_vm_floppy 344f9a075e) last updated 2018/02/02 11:10:37 (GMT -400)
  config file = /Users/deric/.ansible.cfg
  configured module search path = [u'/Users/deric/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/deric/ansible/lib/ansible
  executable location = /Users/deric/ansible/bin/ansible-playbook
  python version = 2.7.14 (default, Sep 25 2017, 09:54:19) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)]
Using /Users/deric/.ansible.cfg as config file
setting up inventory plugins
Set default localhost to localhost
Parsed /Users/deric/ansible/hosts.yml inventory source with yaml plugin
[DEPRECATION WARNING]: vsphere_guest is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.. This feature will be removed in a future release. Deprecation warnings can be disabled by 
setting deprecation_warnings=False in ansible.cfg.
Loading callback plugin default of type stdout, v2.0 from /Users/deric/ansible/lib/ansible/plugins/callback/default.pyc

PLAYBOOK: reconfigure_vm_floppy.yml ********************************************************************************************************************************************************************************************************************************************
1 plays in reconfigure_vm_floppy.yml

PLAY [Reconfigure vm_floppy] ***************************************************************************************************************************************************************************************************************************************************
META: ran handlers

TASK [Reconfigure removable devices to 'client'] *******************************************************************************************************************************************************************************************************************************
task path: /Users/deric/ansible/reconfigure_vm_floppy.yml:7
Using module file /Users/deric/ansible/lib/ansible/modules/cloud/vmware/_vsphere_guest.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: deric
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/deric/.ansible/tmp/ansible-tmp-1518450700.95-133214086404902 `" && echo ansible-tmp-1518450700.95-133214086404902="` echo /Users/deric/.ansible/tmp/ansible-tmp-1518450700.95-133214086404902 `" ) && sleep 0'
<localhost> PUT /Users/deric/.ansible/tmp/ansible-local-2126w0cvFo/tmp5WwJEF TO /Users/deric/.ansible/tmp/ansible-tmp-1518450700.95-133214086404902/_vsphere_guest.py
<localhost> EXEC /bin/sh -c 'chmod u+x /Users/deric/.ansible/tmp/ansible-tmp-1518450700.95-133214086404902/ /Users/deric/.ansible/tmp/ansible-tmp-1518450700.95-133214086404902/_vsphere_guest.py && sleep 0'
<localhost> EXEC /bin/sh -c '/Users/deric/.virtualenvs/ansible-jD3HZhMR/bin/python /Users/deric/.ansible/tmp/ansible-tmp-1518450700.95-133214086404902/_vsphere_guest.py && sleep 0'
[DEPRECATION WARNING]: The 'vsphere_guest' module has been deprecated. Use 'vmware_guest' instead.. This feature will be removed in version 2.9. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
changed: [vm01.lan -> localhost] => {
    "changed": true, 
    "changes": {
        "floppy": {
            "type": "client"
        }
    }, 
    "failed": false, 
    "invocation": {
        "module_args": {
            "cluster": null, 
            "esxi": {
                "datacenter": "datacenter01", 
                "hostname": "esxi01.lan"
            }, 
            "force": false, 
            "from_template": null, 
            "guest": "vm01", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "power_on_after_clone": true, 
            "resource_pool": null, 
            "snapshot_to_clone": null, 
            "state": "reconfigured", 
            "template_src": null, 
            "username": "root", 
            "validate_certs": false, 
            "vcenter_hostname": "vc01.lan", 
            "vm_disk": {}, 
            "vm_extra_config": {}, 
            "vm_hardware": {
                "vm_floppy": {
                    "type": "client"
                }
            }, 
            "vm_hw_version": null, 
            "vm_nic": {}, 
            "vmware_guest_facts": null
        }
    }
}

TASK [Reconfigure vm floppy to 'image'] ****************************************************************************************************************************************************************************************************************************************
task path: /Users/deric/ansible/reconfigure_vm_floppy.yml:25
Using module file /Users/deric/ansible/lib/ansible/modules/cloud/vmware/_vsphere_guest.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: deric
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/deric/.ansible/tmp/ansible-tmp-1518450706.57-185313718759568 `" && echo ansible-tmp-1518450706.57-185313718759568="` echo /Users/deric/.ansible/tmp/ansible-tmp-1518450706.57-185313718759568 `" ) && sleep 0'
<localhost> PUT /Users/deric/.ansible/tmp/ansible-local-2126w0cvFo/tmptHDDY7 TO /Users/deric/.ansible/tmp/ansible-tmp-1518450706.57-185313718759568/_vsphere_guest.py
<localhost> EXEC /bin/sh -c 'chmod u+x /Users/deric/.ansible/tmp/ansible-tmp-1518450706.57-185313718759568/ /Users/deric/.ansible/tmp/ansible-tmp-1518450706.57-185313718759568/_vsphere_guest.py && sleep 0'
<localhost> EXEC /bin/sh -c '/Users/deric/.virtualenvs/ansible-jD3HZhMR/bin/python /Users/deric/.ansible/tmp/ansible-tmp-1518450706.57-185313718759568/_vsphere_guest.py && sleep 0'
changed: [vm01.lan -> localhost] => {
    "changed": true, 
    "changes": {
        "floppy": {
            "image_path": "datastore01/flps/bg-tlb-2.0.img", 
            "type": "image"
        }
    }, 
    "failed": false, 
    "invocation": {
        "module_args": {
            "cluster": null, 
            "esxi": {
                "datacenter": "datacenter01", 
                "hostname": "esxi01.lan"
            }, 
            "force": false, 
            "from_template": null, 
            "guest": "vm01", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "power_on_after_clone": true, 
            "resource_pool": null, 
            "snapshot_to_clone": null, 
            "state": "reconfigured", 
            "template_src": null, 
            "username": "root", 
            "validate_certs": false, 
            "vcenter_hostname": "vc01.lan", 
            "vm_disk": {}, 
            "vm_extra_config": {}, 
            "vm_hardware": {
                "vm_floppy": {
                    "image_path": "datastore01/flps/bg-tlb-2.0.img", 
                    "type": "image"
                }
            }, 
            "vm_hw_version": null, 
            "vm_nic": {}, 
            "vmware_guest_facts": null
        }
    }
}

TASK [Reconfigure removable devices to 'client'] *******************************************************************************************************************************************************************************************************************************
task path: /Users/deric/ansible/reconfigure_vm_floppy.yml:44
Using module file /Users/deric/ansible/lib/ansible/modules/cloud/vmware/_vsphere_guest.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: deric
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/deric/.ansible/tmp/ansible-tmp-1518450714.45-173309074190635 `" && echo ansible-tmp-1518450714.45-173309074190635="` echo /Users/deric/.ansible/tmp/ansible-tmp-1518450714.45-173309074190635 `" ) && sleep 0'
<localhost> PUT /Users/deric/.ansible/tmp/ansible-local-2126w0cvFo/tmpx7zSWk TO /Users/deric/.ansible/tmp/ansible-tmp-1518450714.45-173309074190635/_vsphere_guest.py
<localhost> EXEC /bin/sh -c 'chmod u+x /Users/deric/.ansible/tmp/ansible-tmp-1518450714.45-173309074190635/ /Users/deric/.ansible/tmp/ansible-tmp-1518450714.45-173309074190635/_vsphere_guest.py && sleep 0'
<localhost> EXEC /bin/sh -c '/Users/deric/.virtualenvs/ansible-jD3HZhMR/bin/python /Users/deric/.ansible/tmp/ansible-tmp-1518450714.45-173309074190635/_vsphere_guest.py && sleep 0'
changed: [vm01.lan -> localhost] => {
    "changed": true, 
    "changes": {
        "floppy": {
            "type": "client"
        }
    }, 
    "failed": false, 
    "invocation": {
        "module_args": {
            "cluster": null, 
            "esxi": {
                "datacenter": "datacenter01", 
                "hostname": "esxi01.lan"
            }, 
            "force": false, 
            "from_template": null, 
            "guest": "vm01", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "power_on_after_clone": true, 
            "resource_pool": null, 
            "snapshot_to_clone": null, 
            "state": "reconfigured", 
            "template_src": null, 
            "username": "root", 
            "validate_certs": false, 
            "vcenter_hostname": "vc01.lan", 
            "vm_disk": {}, 
            "vm_extra_config": {}, 
            "vm_hardware": {
                "vm_floppy": {
                    "type": "client"
                }
            }, 
            "vm_hw_version": null, 
            "vm_nic": {}, 
            "vmware_guest_facts": null
        }
    }
}
META: ran handlers
META: ran handlers

PLAY RECAP *********************************************************************************************************************************************************************************************************************************************************************
vm01.lan    : ok=3    changed=3    unreachable=0    failed=0   
```
